### DISCUSSION TITLE
I'm not sure how to use custom classes with CSS state and media queries

### DISCUSSION THREAD

**Discussion by dawidseipold** [muhammadsammy/tailwindcss-classnames#443] [Q&A]
How would you go about creating something like this:

I have two example classes:
```js
export const small = classnames(fontSize('text-sm'));
export const large = classnames(fontSize('text-lg'));
```

And I want to use them depending on the CSS state (hover, md, focus, etc.)

Something like this (obviously the code below isn't correct):
```js
export const title = ({ background }: ITitle) => classnames(typography(small, { md: large }));
```

**Comment by muhammadsammy** [comment]
Why not use TailwindCSS pseudoclasses like so:

```ts
import classnames, { typography } from "tailwindcss-classnames"

const title = classnames(typography("text-sm", "hover:text-lg"))
```
css state methods was supported but it was dropped because the classnames passed to these methods was being purged https://github.com/muhammadsammy/tailwindcss-classnames/issues/13
